### DISCUSSION TITLE
Rover: Network Profiling tab not showing topology view

### DISCUSSION THREAD

**Discussion by p1742000** [apache/skywalking#11104] [General]
Hi team, 

I have a running kubernetes environment and i want network profiling to be enabled by rover. i have followed official documentation to deploy and configure rover with skywalking and istio. (Link :- https://skywalking.apache.org/docs/main/next/en/academy/diagnose-service-mesh-network-performance-with-ebpf/ )

i have configured SW_METER_ANALYZER_ACTIVE_FILES='network-profiling' also as per the official documentation. When i am creating task for network profiling, in rover logs i am getting "key not found : ExtentionConfigJSON " error.

time="2023-07-14T06:42:40Z" level=warning msg="could not execute task, ignored. parsing profiling task failure, command: [key:\"SerialNumber\"  value:\"b4c5e911-c853-4945-8fa3-70d148978066\" key:\"TaskId\"  value:\"282c9b767a3fefe46a28bcb90a391f43cdb0c04cf938e62f38c02d4007df37ff\" key:\"ProcessId\"  value:\"7c13bd91dbfdadd48d959be260602ec68320f29fb64242fd9fee72082b575140,abfb73baa85edd92f8d9daf54882d1223b8dfe458421ade34005c91e7563592c\" key:\"TaskUpdateTime\"  value:\"1689261716574\" key:\"TriggerType\"  value:\"FIXED_TIME\" key:\"TargetType\"  value:\"NETWORK\" key:\"TaskStartTime\"  value:\"1689261716574\" key:\"FixedTriggerDuration\"  value:\"600\"], reason: could not found key: ExtensionConfigJSON" module=profiling

<img width="1276" alt="image" src="https://github.com/apache/skywalking/assets/61838673/c4b54e67-4760-4d85-bfe0-bc52e0b35336">

kindly pls, help me weather anything else i need to add as part of configuration from istio side or OAP side apart given in the official doc. bec, as per documentation i have deployed traffic-generator also..but still in the skywalking UI services showing no load (0.00).

<img width="1781" alt="image" src="https://github.com/apache/skywalking/assets/61838673/2a57e7e8-e84e-4def-a53c-447755ffc7c6">

This is the traffic generator and rover file given in the official doc. that i have followed :- 
https://raw.githubusercontent.com/mrproliu/skywalking-network-profiling-demo/main/resources/traffic-generator.yaml

https://raw.githubusercontent.com/mrproliu/skywalking-network-profiling-demo/main/resources/skywalking-rover.yaml

**Comment by mrproliu** [comment]
Please update your OAP to the latest `9.6.0` version.

**Reply by p1742000** [reply]
Hii, Thanks for your quick reply. For my clarification only asking, wouldn't it be possible to use network profiling in SW-OAP : 9.2.0 as we just recently upgrading skywalking to 9.2.0 in our organization, it is hard to upgrade OAP with 9.6.0 at this point.  
(yeah, surely i will upgrade my OAP to 9.6.0 for testing/POC purpose for surely)

**Reply by mrproliu** [reply]
Ok, then you should be trying on the rover with the v0.3.0 version. It should be worked. 
For more detail about the latest rover version with v0.3.0, [please read this documentation](https://github.com/apache/skywalking-rover/blob/main/CHANGES.md).

**Reply by p1742000** [reply]
Okay, currently i am trying with rover-0.4.0. i will try with rover-0.3.0 (with OAP-9.2.0) and then let you know. Thanks.

**Reply by p1742000** [reply]
Hy @mrproliu , i deployed rover with 0.3.0 version with OAP-9.2.0 only. but getting this below error while trying to create task in network profiling. and no topology is showing.

<img width="1290" alt="image" src="https://github.com/apache/skywalking/assets/61838673/502093a0-99fd-47d4-9d4c-87d697f533f5">

time="2023-07-17T07:58:16Z" level=info msg="detection has reached the properties report factor, forced to report all processes properties" module=process.finder
time="2023-07-17T07:59:16Z" level=info msg="received 1 profiling task: [e79b854e8b393d6c3892193a7c4b63ae5e1cc35b5b682b01aecbdcc84bd5d201]" module=profiling
time="2023-07-17T07:59:16Z" level=info msg="ready to starting profiling task: e79b854e8b393d6c3892193a7c4b63ae5e1cc35b5b682b01aecbdcc84bd5d201" module=profiling.task
time="2023-07-17T07:59:16Z" level=warning msg="found BTF failure: could not found customized BTF file: files/alpine/3.16.1/x86_64/5.15.49-linuxkit.btf" module=tools.btf
time="2023-07-17T07:59:16Z" level=warning msg="executing profiling task failure, taskId: e79b854e8b393d6c3892193a7c4b63ae5e1cc35b5b682b01aecbdcc84bd5d201, reason: field SysAcceptRet: program sys_accept_ret: CO-RE relocations: no BTF found for kernel version 5.15.49-linuxkit: not supported" module=profiling.task

**Reply by mrproliu** [reply]
It looks like not support with your operation system. Could you please execute the "uname-a" to getting the operation system info.

**Reply by p1742000** [reply]
<img width="1005" alt="image" src="https://github.com/apache/skywalking/assets/61838673/5fc231e3-5050-4a9d-97b7-87cf4e1cadf0">

Darwin 4774113-4052c 21.6.0 Darwin Kernel Version 21.6.0: Thu Mar  9 20:08:59 PST 2023; root:xnu-8020.240.18.700.8~1/RELEASE_X86_64 x86_64

**Reply by p1742000** [reply]
After some time rover logs giving this error : proc/process ID/environ : no such file or directory

<img width="1273" alt="image" src="https://github.com/apache/skywalking/assets/61838673/252d7061-3202-4bab-b157-b1b15a89dd9b">

**Reply by mrproliu** [reply]
It looks like you are running in the MacOS. We only support Linux for now. [Please read this documentation.](https://skywalking.apache.org/docs/skywalking-rover/next/en/setup/overview/#prerequisites)

**Reply by p1742000** [reply]
i could share my rover config and application.yaml from OAP side and istio config map from istio side to have more clarification

rover config :- https://github.com/p1742000/rover/blob/main/rover.yaml
OAP application.yaml :- https://github.com/p1742000/rover/blob/main/application.yaml
istio-config :- https://github.com/p1742000/rover/blob/main/istio-config.yaml

Kindly, pls look at rover config, if requires. Thanks.

**Reply by mrproliu** [reply]
This is unrelated to your configuration. Please read the document I mentioned earlier to confirm if your machine supports it.
